bot/__init__.py

```diff
@@ -168,6 +171,46 @@ def _get_word(self) -> str:
         result = self.buffer[self.previous:self.index + (pos+2)]
         self.index += 2
 
+    # Check if a command in the form of `bot.tags['ask'] {= 'whatever'}` was used
+    elif current == "[":
+        # Helper function to remove any characters we don't care about
+        def clean_argument(arg: str) -> str:
+            return arg.strip("[]'\" ")
+
+        # Syntax is `bot.tags['ask']` => mimic `getattr`
+        if self.buffer.endswith("]"):
+            # Key: The first argument, specified `bot.tags[here]`
+            key = clean_argument(self.buffer[self.index:])
+            # note: if not key, this corresponds to an empty argument
+            #       so this should throw / return a SyntaxError ?
+            args = f"\"{key}\""
+            # Use the cog's `get` command.
+            result = self.buffer[self.previous:self.index] + ".get"
+
+        # Syntax is `bot.tags['ask'] = 'whatever'` => mimic `setattr`
+        elif "=" in self.buffer and not self.buffer.endswith("="):
+            equals_pos = self.buffer.find("=")
+            # Key: The first argument, specified `bot.tags[here]`
```

Comment:
some comments, like this one and the one below, would probably benefit from a blank line above them so the code wouldn't feel so cramped in this section. Have a look around for other comments that could lighten up the code density by having a blank line above them. Generally you can put blank lines above any comment, but doing it too often can hurt readability sometimes. Balancing blank lines is a readability problem without a solution, so just use your best judgement.

This is just a food for thought sort of comment, not a requirement. If you think it looks best like this, keep it like this. But do consider it, at least.

Reply:
Thanks for the comment. It's very useful information. Will keep it in mind in the future